release-tools/repack-debian-tarball.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+# This script is used to re-pack the "orig" tarball from the Debian package
+# into a suitable upstream release. There are two changes applied: The Debian
+# tarball contains the directory snapd.upstream/ which needs to become
+# snapd-$VERSION. The Debian tarball contains the vendor/ directory which must
+# be removed from one of those.
+#
+# Example usage, using tarball from the archive or from the image ppa:
+#
+# $ wget https://launchpad.net/ubuntu/+archive/primary/+files/snapd_2.31.2.tar.xz 
+# $ wget https://launchpad.net/~snappy-dev/+archive/ubuntu/image/+files/snapd_2.32.1.tar.xz
+#
+# $ repack-debian-tarball.sh snapd_2.31.2.tar.xz
+#
+# This will produce three files that need to be added to the github release page:
+#
+# - snapd_2.31.2.no-vendor.tar.xz
+# - snapd_2.31.2.vendor.tar.xz
+# - snapd_2.31.2.only-vendor.xz
+set -xue
+
+# Get the filename from argv[1]
+debian_tarball="$1"
+if [ "$debian_tarball" = "" ]; then
+	echo "Usage: repack-debian-tarball.sh <snapd-debian-tarball>"
+	exit 1
+fi
+
+if [ ! -f "$debian_tarball" ]; then
+	echo "cannot operate on $debian_tarball, no such file"
+	exit 1
+fi
+
+# Extract the upstream version from the filename.
+# For example: snapd_2.31.2-1.tar.xz => 2.32.2
+upstream_version="$(echo "$debian_tarball" | cut -d _ -f 2 | sed -e 's/\.tar\..*//')"
```

Comment:
Either the comment or the code is wrong. Given `snapd_2.31.2-1.tar.xz`, the code will produce `2.31.2-1`. Maybe you want: `echo "$debian_tarball" | cut -d_ -f2 | cut -d- -f1`

Reply:
Oh indeed, the comment is wrong.